### DISCUSSION TITLE
Can't install gatsby-cli, does not support 18.7.0 version and cb.apply is not a function

### DISCUSSION THREAD

**Discussion by GregPlaya** [gatsbyjs/gatsby#36449] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Following directions for installing Gatsby for Windows 10 here:
https://www.gatsbyjs.com/docs/tutorial/part-0/

Please help!!!  I'm new to this and need to edit a website on Gatsby today.  
I get several errors when I run:  npm install -g gatsby-cli

npm WARN npm npm does not support Node.js v18.7.0  (I have also tried v16.17.0)
and
npm ERR! cb.apply is not a function

![Gatsby-install-problem](https://user-images.githubusercontent.com/110426691/185988382-ddc9291a-abbd-406b-ae03-091ce8baa730.jpg)

log
0 info it worked if it ends with ok
1 warn npm npm does not support Node.js v18.7.0
2 warn npm You should probably upgrade to a newer version of node as we
3 warn npm can't make any promises that npm will work with this version.
4 warn npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11, 12.
5 warn npm You can find the latest version at https://nodejs.org/
6 verbose cli [
6 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
6 verbose cli   'C:\\Users\\Greg\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js',
6 verbose cli   'install',
6 verbose cli   '-g',
6 verbose cli   'gatsby-cli'
6 verbose cli ]
7 info using npm@6.9.0
8 info using node@v18.7.0
9 verbose npm-session 2e23ca243f8971c2
10 silly install loadCurrentTree
11 silly install readGlobalPackageData
12 http fetch GET 304 https://registry.npmjs.org/gatsby-cli 391ms (from cache)
13 silly pacote tag manifest for gatsby-cli@latest fetched in 444ms
14 verbose stack TypeError: cb.apply is not a function
14 verbose stack     at C:\Users\Greg\AppData\Roaming\npm\node_modules\npm\node_modules\graceful-fs\polyfills.js:285:20
14 verbose stack     at FSReqCallback.oncomplete (node:fs:212:5)
15 verbose cwd C:\Users\Greg
16 verbose Windows_NT 10.0.19043
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Greg\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "gatsby-cli"
18 verbose node v18.7.0
19 verbose npm  v6.9.0
20 error cb.apply is not a function
21 verbose exit [ 1, true ]

### Reproduction Link

https://github.com/GregPlaya/sandbox-solana

### Steps to Reproduce

1. run:  npm install -g gatsby-cli
2. check for version: gatsby --version
3.
...

### Expected Result

installs?

### Actual Result

errors

### Environment

```shell
I don't know how to do this??
```

### Config Flags

None

**Comment by tyhopp** [comment]
This seems like an issue with your node/npm installations. Can you reproduce this on other machines or Docker containers?

**Reply by GregPlaya** [reply]
Thanks for the reply, I could try on my personal laptop but would rather not.  Need to get it on my workstation.   Should I uninstall node or npm? and reinstall? and anything else?  And which versions should I use? (I'm on Windows 10)   I took over a website from someone who transferred it to this and I just need to do some text edits to the web page.  Maybe there is a way to edit Gatsby pages directly?  (without Github and all of this -  this seems like a lot for a 2 page website.)
